qutip/random_objects.py

```diff
@@ -187,18 +187,20 @@ def _rand_herm_sparse(N, density, pos_def):
                              in np.random.choice(N*N,
                                                  num_elems,
                                                  replace=False)])
-    M = sp.coo_matrix((data, (row_idx,col_idx)),
-                      dtype=complex, shape=(N,N)).tocsr()
+    M = sp.coo_matrix((data, (row_idx, col_idx)),
+                      dtype=complex, shape=(N, N))
     M = 0.5 * (M + M.conj().transpose())
     if pos_def:
-        M.setdiag(np.abs(M.diagonal())+np.sqrt(2)*N)
+        M = M.tolil()
+        M.setdiag(np.abs(M.diagonal()) + np.sqrt(2)*N)
```

Comment:
What's the reason for converting to lil?  I'd have expected that the cost of converting to lil, and the increased cost of the lil->csr conversion (coo->csr is optimised in scipy) dominated any other effects.